src/coreclr/src/jit/lowerarmarch.cpp

```diff
@@ -706,12 +706,18 @@ void Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cmpOp)
 
     if ((baseType == TYP_FLOAT) && (simdSize == 12))
     {
-        // For TYP_SIMD12 we need to clear the upper bits and can't assume their value
+        // For TYP_SIMD12 the value of the upper bits is unknown but shouldn't impact
+        // the comparison. So, we should always pretend the bits were equivalent to
+        // ensure that the final result is correct. In the case of equality, the unsigned
+        // min across will always treat -1 (all bits set) as higher and so it will be
+        // preserved if the other three elements were also equal. In the case of inequality
+        // it will likewise be handled and won't cause an otherwise equivalent vector from
+        // being treated as not equal.
```

Comment:
I found this just a bit confusing. I might reword as
```suggestion
        // For TYP_SIMD12 we don't want the upper bits to participate in the comparison. So, we will insert all ones
        // into those bits of the result, "as if" the upper bits are equal. Then if all lower bits are equal, we get the 
        // expected all-ones result, and will get the expected 0's only where there are non-matching bits.
```
Or something along those lines.